### PR TITLE
rgw: change default chunk size to 4MB

### PR DESCRIPTION
**Pull request by yehudasa**
1. Fixes: http://tracker.ceph.com/issues/18621
change default chunk size to 4MB

2. Fixes: http://tracker.ceph.com/issues/18622
optimize first writes on the expense of overwrites

3. Fixes: http://tracker.ceph.com/issues/18623
configurable min and max write window sizes.

Signed-off-by: Yehuda Sadeh <yehuda@redhat.com>

### CODE REVIEW

File: src/rgw/rgw_rados.h [hunk 1]
```diff
@@ -3400,9 +3412,10 @@ class RGWPutObjProcessor_Aio : public RGWPutObjProcessor
   int handle_obj_data(rgw_obj& obj, bufferlist& bl, off_t ofs, off_t abs_ofs, void **phandle, bool exclusive);
 
 public:
-  int throttle_data(void *handle, const rgw_obj& obj, bool need_to_wait);
+  int prepare(RGWRados *store, string *oid_rand);
+  int throttle_data(void *handle, const rgw_obj& obj, uint64_t size, bool need_to_wait);
 
-  RGWPutObjProcessor_Aio(RGWObjectCtx& obj_ctx, RGWBucketInfo& bucket_info) : RGWPutObjProcessor(obj_ctx, bucket_info), max_chunks(RGW_MAX_PENDING_CHUNKS), obj_len(0) {}
```
Comment (prallabh) [inline src/rgw/rgw_rados.h]: Should we also remove RGW_MAX_PENDING_CHUNKS from rgw_common.h?

**Comment by mattbenjamin** [comment]
This passed a short suite run Fri., one test failed due to ntp.
http://pulpito.ceph.com/mbenjamin-2017-02-04_14:08:21-rgw-yehudasa-wip-18621---basic-smithi/

File: src/rgw/rgw_rados.cc [hunk 2]
```diff
@@ -6424,7 +6435,7 @@ int RGWRados::Object::Write::_do_write_meta(uint64_t size, uint64_t accounted_si
     if (r == -EEXIST && assume_noent) {
       target->invalidate_state();
     }
-    goto done_cancel;
+    return r;
```
Comment (tianshan) [inline src/rgw/rgw_rados.cc]: maybe we still should goto done_cancel if return -ECANCELED or -ENOENT ?

**Comment by mattbenjamin** [comment]
http://pulpito.ceph.com/mbenjamin-2017-02-06_17:42:49-rgw-wip-yehudasa-18621---basic-smithi/

**Comment by mattbenjamin** [comment]
@yehudasa this run was nearly all green--one dead agent task;  I think it's ready

**Pull request merged**